tests/functional/test_cluster_functions.py

```diff
@@ -9,443 +9,222 @@
-        with pytest.raises(coreapi.exceptions.ErrorMessage) as e:
-            client.cluster.create(prototype_id='', name=utils.random_string())
-        with allure.step('Check error about valid integer'):
-            assert e.value.error.title == '400 Bad Request'
-            assert e.value.error['prototype_id'][0] == 'A valid integer is required.'
-        steps.delete_all_data(client)
 
-    def test_shouldnt_create_cluster_when_proto_is_string(self, client):
-        with pytest.raises(coreapi.exceptions.ErrorMessage) as e:
-            client.cluster.create(prototype_id=utils.random_string(), name=utils.random_string())
-        with allure.step('Check error about valid integer'):
-            assert e.value.error.title == '400 Bad Request'
-            assert e.value.error['prototype_id'][0] == 'A valid integer is required.'
-        steps.delete_all_data(client)
+def _create_service(cluster_: Cluster, prototype_id):
+    """Workaround to create service without retrieving prototype"""
+    cluster_._subcall(
+        "service", "create", prototype_id=prototype_id, cluster_id=cluster_.id
+    )
 
-    def test_shouldnt_create_cluster_if_proto_not_find(self, client):
-        with pytest.raises(coreapi.exceptions.ErrorMessage) as e:
-            client.cluster.create(prototype_id=random.randint(900, 950), name=utils.random_string())
-        with allure.step('Check error about prototype doesn\'t exist'):
-            err.PROTOTYPE_NOT_FOUND.equal(e, 'prototype doesn\'t exist')
-        steps.delete_all_data(client)
 
-    def test_shouldnt_create_cluster_wo_name(self, client):
-        prototype = get_random_cluster_prototype(client)
-        with pytest.raises(coreapi.exceptions.ErrorMessage) as e:
-            client.cluster.create(prototype_id=prototype['id'], name='')
-        with allure.step('Check error about blank field'):
-            assert e.value.error.title == '400 Bad Request'
-            assert e.value.error['name'] == ['This field may not be blank.']
-        steps.delete_all_data(client)
-
-    def test_shoulndt_create_cluster_when_desc_is_null(self, sdk_client_fs: ADCMClient):
-        bundle = sdk_client_fs.upload_from_fs(get_data_dir(__file__, 'cluster_simple'))
-        with pytest.raises(coreapi.exceptions.ErrorMessage) as e:
-            bundle.cluster_create(name=utils.random_string(), description='')
-        with allure.step('Check error about blank field'):
-            assert e.value.error.title == '400 Bad Request'
-            assert e.value.error['description'] == ["This field may not be blank."]
-
-    def test_get_cluster_list(self, sdk_client_fs: ADCMClient):
-        bundle = sdk_client_fs.upload_from_fs(get_data_dir(__file__, 'cluster_simple'))
-        expectedlist = []
-        actuallist = []
+def _patch_cluster(cluster_: Cluster, **kwargs):
+    """Call partial_update endpoint (does not implemented in adcm_client)"""
+    cluster_._subcall("partial_update", **kwargs)
```

Comment:
Why are you using private methods? If you need methods that are not in the client, then you need to ask us to add these methods, and not use private ones.

Reply:
I'll create feature request and replace by added method after release.
All of the extracted methods with private methods calls are possible candidates to implement in adcm_client, expect some cases like in the comment above.